src/flypad-backend/src/Aircraft/AircraftProcedures.h

```diff
@@ -237,29 +238,47 @@ class AircraftProcedures {
 
   template<std::size_t N1, std::size_t N2>
   static constexpr void doInsertProcedures(std::array<const ProcedureStep*, N1>& dest, const std::array<ProcedureStep, N2>& src, const std::size_t offset) {
-    std::transform(begin(src), end(src), begin(dest) + offset * N2, [](const auto& procedure) {
+    std::transform(begin(src), end(src), begin(dest) + offset, [](const auto& procedure) {
       return &procedure;
     });
   }
 
   template<class Tuple, std::size_t S, std::size_t... I>
   static constexpr void doInsertProcedures(std::array<const ProcedureStep*, S>& dest, const Tuple& procedures, std::index_sequence<I...>) {
-    (doInsertProcedures(dest, std::get<I>(procedures), I), ...);
+    (doInsertProcedures(dest, std::get<I>(procedures), (I == 0 ? 0 : I - 1) * std::get<(I == 0 ? 0 : I - 1)>(procedures).size()), ...);
+  }
+
+  template<std::size_t... I>
+  static constexpr void doInsertProcedures(auto& dest, const auto& sources, const auto& offsets, std::index_sequence<I...>) {
+    (doInsertProcedures(dest, std::get<I>(sources), offsets[I]), ...);
+  }
+
+  template<class Tuple, std::size_t... I>
+  static constexpr std::array<std::size_t, sizeof...(I) + 1> getOffsets(const Tuple& procedures, std::index_sequence<I...>) {
+    std::array sizes = {(std::size_t)0, std::get<I>(procedures).size()...};
+    std::array<size_t, sizeof...(I) + 1> offsets{};
+    std::inclusive_scan(begin(sizes), end(sizes), begin(offsets));
```

Comment:
What you implement here is essentially `std::exclusive_scan`:
```suggestion
static constexpr std::array<std::size_t, sizeof...(I)> getOffsets(const Tuple& procedures, std::index_sequence<I...>) {
    std::array sizes = {std::get<I>(procedures).size()...};
    std::array<size_t, sizeof...(I)> offsets{};
    std::exclusive_scan(begin(sizes), end(sizes), begin(offsets), 0);
```